tests/unit/plugins/modules/system/test_sapcontrol.py

```diff
@@ -0,0 +1,129 @@
+            "sysnr": "01",
+            "function": "GetProcessList"
+        })
+        with self.assertRaises(AnsibleFailJson) as result:
+            self.module.Client.side_effect = Mock(side_effect=Exception('Test'))
+            self.module.main()
+        self.assertEqual(result.exception.args[0]['msg'], 'Something went wrong connecting to the SAPCONTROL SOAP API.')
+
+    def test_error_port_sysnr(self):
+        """tests fail multi provide parameters"""
+
+        set_module_args({
+            "hostname": "192.168.8.15",
+            "sysnr": "01",
+            "port": "50113",
+            "function": "GetProcessList"
+        })
+        with self.assertRaises(AnsibleFailJson) as result:
+            self.module.main()
+        self.assertEqual(result.exception.args[0]['msg'], 'parameters are mutually exclusive: sysnr|port')
+
+    def test_error_missing_force(self):
+        """tests fail missing force"""
+
+        set_module_args({
+            "hostname": "192.168.8.15",
+            "sysnr": "01",
+            "function": "Stop"
+        })
+
+        with self.assertRaises(AnsibleFailJson) as result:
+            self.module.main()
+        self.assertEqual(result.exception.args[0]['msg'], 'Stop function requires force: True')
+
+    def test_success_sysnr(self):
+        """test success with sysnr"""
+
+        set_module_args({
+            "hostname": "192.168.8.15",
+            "sysnr": "01",
+            "function": "GetProcessList"
+        })
+        with patch.object(self.module, 'recursive_dict') as ret:
+            ret.return_value = {'item': [{'name': 'hdbdaemon', 'value': '1'}]}
+            with self.assertRaises(AnsibleExitJson) as result:
+                self.module.main()
+        self.assertEqual(result.exception.args[0]['out'], [{'item': [{'name': 'hdbdaemon', 'value': '1'}]}])
+
+    def test_success_port(self):
+        """test success with port"""
+
+        set_module_args({
+            "hostname": "192.168.8.15",
+            "port": "50113",
+            "function": "GetProcessList"
+        })
+        with patch.object(self.module, 'recursive_dict') as ret:
+            ret.return_value = {'item': [{'name': 'hdbdaemon', 'value': '1'}]}
+            with self.assertRaises(AnsibleExitJson) as result:
+                self.module.main()
+        self.assertEqual(result.exception.args[0]['out'], [{'item': [{'name': 'hdbdaemon', 'value': '1'}]}])
+
+    def test_success_string(self):
+        """test success with sysnr"""
+
+        set_module_args({
+            "hostname": "192.168.8.15",
+            "sysnr": "01",
+            "function": "ParameterValue",
+            "parameter": "ztta/short_area"
+        })
+        with patch.object(self.module, 'connection') as ret:
+            ret.return_value = '1600000'
+            with self.assertRaises(AnsibleExitJson) as result:
+                self.module.main()
+        self.assertEqual(result.exception.args[0]['out'], ['1600000'])
```

Comment:
just curious; What does the ParameterValue ztta/short_area describe?

Reply:
I dont't know it was the first outputted key :-D